File: app/src/main/java/life/plank/juna/zone/interfaces/EngagementInfoTilesToolbar.java
package life.plank.juna.zone.interfaces;

import android.support.annotation.DrawableRes;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;

public interface EngagementInfoTilesToolbar {

    void setLeagueLogo(String logoUrl);

    void setLeagueLogo(@DrawableRes int resource);

    void setPeopleCount(String peopleCount);

    void setCommentCount(String commentsCount);

    void setBoardTitle(String boardTitle);

    void showLock(boolean showLock);

    boolean isFollowing();

    void setFollowing(boolean isFollowing);

    TabLayout getInfoTilesTabLayout();

    void setupWithViewPager(ViewPager viewPager);
}

File: app/src/main/java/life/plank/juna/zone/interfaces/EngagementInfoTilesToolbar.kt
package life.plank.juna.zone.interfaces

import androidx.annotation.DrawableRes
import com.google.android.material.tabs.TabLayout

interface EngagementInfoTilesToolbar {

    fun getInfoTilesTabLayout(): TabLayout

    fun setLeagueLogo(logoUrl: String)

    fun setLeagueLogo(@DrawableRes resource: Int)

    fun setPeopleCount(peopleCount: String)

    fun setPostCount(postCount: String)

    fun setBoardTitle(boardTitle: String)

    fun showLock(showLock: Boolean)

    fun setupWithViewPager(viewPager: androidx.viewpager.widget.ViewPager, defaultSelection: Int)
}